packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -257,56 +250,51 @@ function FiatExchangeAmount({ route }: Props) {
       >
         <View style={styles.amountInputContainer}>
           <View>
-            <Text style={styles.exchangeBodyText}>
-              {isCusdCashIn ? t('global:amount') : t('amountCelo')}
-            </Text>
+            <Text style={styles.exchangeBodyText}>{`${t('global:amount')} (${
+              currency === Currency.Celo ? 'CELO' : currency
+            })`}</Text>
           </View>
           <TextInput
             autoFocus={true}
             keyboardType={'decimal-pad'}
             onChangeText={onChangeExchangeAmount}
-            value={inputAmount.length > 0 ? `${inputCurrencySymbol}${inputAmount}` : undefined}
+            value={inputAmount.length > 0 ? `${currencySymbol}${inputAmount}` : undefined}
             placeholderTextColor={colors.gray3}
-            placeholder={`${inputCurrencySymbol}0`}
+            placeholder={`${currencySymbol}0`}
             style={[
               styles.currencyInput,
-              isCusdCashIn ? styles.dollarCurrencyColor : styles.celoCurrencyColor,
+              currency === Currency.Celo ? styles.celoCurrencyColor : styles.fiatCurrencyColor,
             ]}
             testID="FiatExchangeInput"
           />
         </View>
         <LineItemRow
           textStyle={styles.subtotalBodyText}
           title={
-            <Trans
-              i18nKey={isCusdCashIn ? 'celoDollarsAt' : 'inputSubtotal'}
-              ns={Namespaces.fiatExchangeFlow}
-            >
-              {isCusdCashIn ? 'Celo Dollars @ ' : 'Subtotal @ '}
-              <CurrencyDisplay
-                amount={isCusdCashIn ? oneDollarAmount : oneCeloAmount}
-                showLocalAmount={true}
-              />
+            <Trans>
+              {`${t(displayCurrencyKey)} @ `}
+              <CurrencyDisplay amount={oneUnitAmount(currency)} showLocalAmount={true} />
             </Trans>
           }
           amount={
             <CurrencyDisplay
               amount={{
-                value: dollarAmount,
-                currencyCode: Currency.Dollar,
+                value: inputAmount,
+                currencyCode: currency,
               }}
-              hideSymbol={isCusdCashIn}
-              showLocalAmount={!isCusdCashIn}
+              showLocalAmount={true}
             />
           }
         />
       </KeyboardAwareScrollView>
-      {isCusdCashIn && (
-        <Text style={styles.disclaimerCeloDollars}>{t('disclaimerCeloDollars')}</Text>
+      {currency !== Currency.Celo && (
+        <Text style={styles.disclaimerFiat}>
+          {t('disclaimerFiat', { currency: t(displayCurrencyKey) })}
+        </Text>
       )}
       <Button
         onPress={onPressContinue}
-        showLoading={exchangeRatePair === null}
+        showLoading={exchangeRates.cUSD === null}
```

Comment:
This works because the string value of `Currency.Dollar` is `cUSD`, but I think it's safer to do `exchangeRates[Currency.Dollar]`, which is resistant to changes to the string value of the enum and if we ever change anything related to the enum it will start throwing a build error here so we know we have to fix it here.